services/src/main/java/org/keycloak/authentication/authenticators/x509/CertificateValidator.java

```diff
@@ -374,15 +393,15 @@ private Collection<X509CRL> loadCRLFromFile(CertificateFactory cf, String relati
                         }
                         try (FileInputStream is = new FileInputStream(f.getAbsolutePath())) {
                             X509CRL crl = loadFromStream(cf, is);
-                            return Collections.singleton(crl);
+                            return crl;
                         }
                     }
                 }
             }
             catch(IOException ex) {
                 logger.errorf(ex.getMessage());
             }
-            return Collections.emptyList();
+            return null;
         }
         private X509CRL loadFromStream(CertificateFactory cf, InputStream is) throws IOException, CRLException {
```

Comment:
Ops, one comment was no save. It seems you don't need to wrap the `InputStream` into a `DataInputStream`. 
```java
        private static X509CRL loadFromStream(CertificateFactory cf, InputStream is) throws IOException, CRLException {
            return (X509CRL) cf.generateCRL(is);
        }

```

Reply:
Done! I have also checked all of them closes the InputStream with a try.